grid2op/tests/test_ma.py

```diff
@@ -0,0 +1,567 @@
+        
+    def test_interco(self):
+        
+        self.ma_env = MultiAgentEnv(self.env, self.action_domains, _add_to_name="test_interco")
+        # Tests on interconnections with known values for every agent
+        mask_interco_agent0 = np.array([False, False, False, False, False, False, False, False, False,
+                                False, False, False, False, False, False,  True,  True,  True,
+                                False, False])
+        # We compare the interconnection masks with known values for every agent
+        assert (self.ma_env._action_domains['agent_0']['mask_interco'] == mask_interco_agent0).all()
+        assert (self.ma_env._action_domains['agent_1']['mask_interco'] == mask_interco_agent0).all()
+        interco_is_origin_agent0 = np.array([True, True, True])
+        # We compare the interco_is_origin masks with known values for every agent
+        # interco_is_origin tells whether the corresponding interco is a line_or or not
+        assert (self.ma_env._action_domains['agent_0']['interco_is_origin'] == interco_is_origin_agent0).all()
+        assert (self.ma_env._action_domains['agent_1']['interco_is_origin'] == ~interco_is_origin_agent0).all()
+        
+        # In the two-agent case, they must have the same number of interconnections
+        assert self.ma_env._subgrids_cls['action']['agent_0'].n_interco == self.ma_env._subgrids_cls['action']['agent_1'].n_interco
+        
+        # A line in the original grid is either a line in 
+        # one subgrid (if both its end are in the subdomain) or an interconnection. 
+        # By summing all lines in both subdomains and the interconnection, 
+        # we should get the total number of lines.
+        assert self.ma_env._subgrids_cls['action']['agent_0'].n_line \
+             + self.ma_env._subgrids_cls['action']['agent_1'].n_line \
+             + self.ma_env._subgrids_cls['action']['agent_0'].n_interco == self.env.n_line
+                
+
+    def test_build_subgrid_obj(self):
+        """test the MultiAgentEnv._build_subgrid_obj_from_domain"""
+        
+        # 1
+        action_domains = {
+            'agent_0' : [0,1,2,3, 4],
+            'agent_1' : [5,6,7,8,9,10,11,12,13]
+        }
+        ma_env = MultiAgentEnv(self.env, action_domains, _add_to_name="test_build_subgrid_obj")
+        
+        # We compare the number of generators for every agents' subgrids with known values
+        assert ma_env._subgrids_cls['action']['agent_0'].n_gen == 3
+        assert ma_env._subgrids_cls['action']['agent_1'].n_gen == 3
+        
+        self.checks(ma_env, action_domains)
+        
+        # We compare the interconnection original line ids with known values for every agent
+        assert (ma_env._subgrids_cls['action']['agent_0'].interco_to_lineid == np.array([15,16,17])).all()
+        assert (ma_env._subgrids_cls['action']['agent_1'].interco_to_lineid == np.array([15,16,17])).all()
```

Comment:
same, why still copy paste ? (twice [15, 16, 17])

You should avoid at all cost all kind of copy paste when you code, that's really important

Reply:
Ok, changed it